internal/saptune/saptune.go

```diff
@@ -0,0 +1,95 @@
+package saptune
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/tidwall/gjson"
+	"golang.org/x/mod/semver"
+
+	"github.com/trento-project/workbench/internal/support"
+)
+
+const minimalSaptuneVersion = "v3.1.0"
+
+type Saptune interface {
+	CheckVersionSupport(ctx context.Context) error
+	ApplySolution(ctx context.Context, solution string) error
+	GetAppliedSolution(ctx context.Context) (string, error)
+}
+
+type saptuneClient struct {
+	executor support.CmdExecutor
+	logger   *logrus.Entry
+}
+
+func NewSaptuneClient(
+	executor support.CmdExecutor,
+	logger *logrus.Entry,
+) Saptune {
+	return &saptuneClient{
+		executor: executor,
+		logger:   logger,
+	}
+}
+
+func (saptune *saptuneClient) CheckVersionSupport(ctx context.Context) error {
+	versionOutput, err := saptune.executor.Exec(ctx, "rpm", "-q", "--qf", "%{VERSION}", "saptune")
+	if err != nil {
+		return fmt.Errorf(
+			"could not get the installed saptune version: %w",
+			err,
+		)
+	}
+
+	detectedVersion := string(versionOutput)
+
+	if supported := isSaptuneVersionSupported(detectedVersion); !supported {
+		return fmt.Errorf(
+			"saptune version not supported, installed: %s, minimum supported: %s",
+			detectedVersion,
+			minimalSaptuneVersion,
+		)
+	}
+
+	saptune.logger.Debugf("installed saptune version: %s", detectedVersion)
+
+	return nil
+}
+
+func (saptune *saptuneClient) GetAppliedSolution(ctx context.Context) (string, error) {
+	solutionAppliedOutput, err := saptune.executor.Exec(ctx, "saptune", "--format", "json", "solution", "applied")
+	if err != nil {
+		return "", fmt.Errorf("could not call saptune solution applied: %w", err)
+	}
+	if err != nil {
```

Comment:
You forgot to remove this `if`

Reply:
🙈  c4ae86a6e7a93df0753e915066470dbc7bb11618